tools/automation/cli_linter/rules/help_rules.py

```diff
@@ -33,3 +33,14 @@ def unrecognized_help_parameter_rule(linter, help_entry):
             violations.append(param_help_name)
     if violations:
         raise RuleError('The following parameter help names are invalid: {}'.format(' | '.join(violations)))
+
+@help_file_entry_rule
+def faulty_help_example_rule(linter, help_entry):
+    violations = []
+    for index, example in enumerate(linter.get_help_entry_examples(help_entry)):
+        if 'az '+ help_entry not in example.get('text', ''):
```

Comment:
Wouldn't `startswith` be sufficient and more appropriate here?  Also, do you consider multi-step examples where the target command might not be the first?

Reply:
`startswith` is not sufficient exactly because of multi-step examples.
Ex: https://github.com/Azure/azure-cli/blob/dev/src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/_help.py#L71-L79

Example text may start with a different command or setting of variable, etc.

Reply:
I was thinking you'd step through each line of each example text to find at least one line that starts with the command, but "in" accomplishes the same task I suppose.